sdk/src/derivation_path.rs

```diff
@@ -118,67 +119,82 @@ impl fmt::Debug for DerivationPath {
     }
 }
 
-fn bip44_solana_indexes() -> Vec<ChildIndex> {
-    vec![
-        ChildIndex::Hardened(BIP_44_PURPOSE),
-        ChildIndex::Hardened(BIP_44_COIN_SOL),
-    ]
+pub trait Bip44 {
+    const PURPOSE: u32 = 44;
+    const COIN: u32;
+
+    fn base_indexes(&self) -> Vec<ChildIndex> {
+        vec![
+            ChildIndex::Hardened(Self::PURPOSE),
+            ChildIndex::Hardened(Self::COIN),
+        ]
+    }
+}
+
+pub struct Solana;
```

Comment:
Do you think we're going to end up having to maintain a bunch of these?

Reply:
I hope not. But maybe we can set up a framework for `DerivationPath` callers (eg `clap-utils`) that pulls from a registry of `impl Bip44` definitions...?

Reply:
I wouldn't put any other than this one in sdk, for sure